pkg/meta/parse.go

```diff
@@ -44,13 +45,19 @@ func ParseStorage(metaDB mTypes.MetaDB, storeController storage.StoreController,
 func ParseRepo(repo string, metaDB mTypes.MetaDB, storeController storage.StoreController, log log.Logger) error {
 	imageStore := storeController.GetImageStore(repo)
 
+	var lockLatency time.Time
+
+	imageStore.Lock(&lockLatency)
+
 	indexBlob, err := imageStore.GetIndexContent(repo)
 	if err != nil {
 		log.Error().Err(err).Str("repository", repo).Msg("load-repo: failed to read index.json for repo")
 
 		return err
 	}
 
+	imageStore.Unlock(&lockLatency)
```

Comment:
We should defer and unlock even in case of error.

Reply:
I talked with Laurentiu, and it looks like in this case we want to unlock right away, not when the function exits.